extractor/src/test/java/org/schabi/newpipe/extractor/services/peertube/PeertubeChannelTabExtractorTest.java

```diff
@@ -0,0 +1,212 @@
+        @Override
+        public void testServiceId() {
+            assertEquals(PeerTube.getServiceId(), extractor.getServiceId());
+        }
+
+        @Test
+        @Override
+        public void testName() throws Exception {
+            assertEquals(ChannelTabs.PLAYLISTS, extractor.getName());
+        }
+
+        @Test
+        @Override
+        public void testId() throws ParsingException {
+            assertEquals("video-channels/lqdn_channel@video.lqdn.fr", extractor.getId());
+        }
+
+        @Test
+        @Override
+        public void testUrl() throws ParsingException {
+            assertEquals("https://framatube.org/video-channels/lqdn_channel@video.lqdn.fr/video-playlists",
+                    extractor.getUrl());
+        }
+
+        @Test
+        @Override
+        public void testOriginalUrl() throws Exception {
+            assertEquals("https://framatube.org/video-channels/lqdn_channel@video.lqdn.fr/video-playlists",
+                    extractor.getOriginalUrl());
+        }
+
+        @Test
+        @Override
+        public void testRelatedItems() throws Exception {
+            defaultTestRelatedItems(extractor);
+        }
+
+        @Override
+        public void testMoreRelatedItems() throws Exception {
+            defaultTestMoreItems(extractor);
+        }
+    }
+
+    static class Channels implements BaseListExtractorTest {
+        private static PeertubeChannelTabExtractor extractor;
+
+        @BeforeAll
+        static void setUp() throws IOException, ExtractionException {
+            NewPipe.init(DownloaderTestImpl.getInstance());
+            extractor = (PeertubeChannelTabExtractor)
+                    PeerTube.getChannelTabExtractorFromIdAndBaseUrl("accounts/framasoft",
+                            ChannelTabs.CHANNELS, "https://framatube.org");
+            extractor.fetchPage();
+        }
+
+        @Test
+        @Override
+        public void testServiceId() {
+            assertEquals(PeerTube.getServiceId(), extractor.getServiceId());
+        }
+
+        @Test
+        @Override
+        public void testName() throws Exception {
+            assertEquals(ChannelTabs.CHANNELS, extractor.getName());
+        }
+
+        @Test
+        @Override
+        public void testId() throws ParsingException {
+            assertEquals("accounts/framasoft", extractor.getId());
+        }
+
+        @Test
+        @Override
+        public void testUrl() throws ParsingException {
+            assertEquals("https://framatube.org/accounts/framasoft/video-channels",
+                    extractor.getUrl());
+        }
+
+        @Test
+        @Override
+        public void testOriginalUrl() throws Exception {
+            assertEquals("https://framatube.org/accounts/framasoft/video-channels",
+                    extractor.getOriginalUrl());
+        }
+
+        @Test
+        @Override
+        public void testRelatedItems() throws Exception {
+            defaultTestRelatedItems(extractor);
```

Comment:
Maybe the `defaultTest(Related|More)Items` should take a second parameter that would allow making sure all items are of a specific type (e.g. in this case `ChannelInfoItem`)